R/eatModel.r

```diff
@@ -82,42 +82,68 @@ getConquestVersion <- function(path.conquest, path.temp, asDate = TRUE){
-           if (!nrow(calibT2[["itempars"]]) < nrow(bistaTransfT2[["itempars"]])) { stop("Mismatch between 'calibT2' and 'bistaTransfT2'. \n")}
-           idT1<- unique(bistaTransfT1[["all.Names"]][which(bistaTransfT1[["all.Names"]][,"par"] == "ID"),"derived.par"])
-           idT2<- unique(bistaTransfT2[["all.Names"]][which(bistaTransfT2[["all.Names"]][,"par"] == "ID"),"derived.par"])
-           stopifnot(length(idT1)==1, length(idT2)==1)
-           if ( idT1 != idT2 ) {
-                warning(paste0("ID variables do not match between t1 and t2. ID for t1: '",idT1,"'. ID for t2: '",idT2,"'. \n    IDs will be unified with '",idT1,"'."))
-                recStat <- paste ( "'", idT2 , "' = '", idT1, "'", sep="")
-                colnames ( bistaTransfT2[["personpars"]] ) <- car::recode ( colnames ( bistaTransfT2[["personpars"]] ), recStat)
-           }
-           lc  <- colnames( calibT2[["personpars"]] ) [grep("^linking", colnames(calibT2[["personpars"]]) )]
-           if(length(lc)==0) { stop("No columns with linking error information found in 'calibT2'.\n")}
-           lcn <- paste("trend", eatTools::removePattern(string = lc, pattern = "linking"), sep="")
-           colnames( calibT2[["personpars"]] ) [grep("^linking", colnames(calibT2[["personpars"]]) )] <- lcn
-           merg<- c("group", "imp", "traitLevel", "dimension")
-           frms<- list ( calibT2=calibT2, bistaTransfT1=bistaTransfT1, bistaTransfT2=bistaTransfT2 )
-           toM <- unique(unlist(lapply ( names(frms), FUN = function ( l.Name ) {
-                  l    <- frms[[l.Name]]
-                  drin <- merg %in% colnames(l[["personpars"]])
-                  fehlt<- merg[which(drin==FALSE)]
-                  if (!all(drin == TRUE)) { warning(paste0("Column(s) '",paste(fehlt, collapse = "', '"), "' are unexpectedly missing in '",l.Name,"'."))}
-                  keep <- merg[which(drin==TRUE)]
-                  return(keep)})))
-           if(length(toM)==0) { stop("Merging impossible.\n")}
-           red <- calibT2[["personpars"]][,c(toM,  lcn)]
-           red <- red[!duplicated(red),]
-           dat1<- data.frame ( trend = "T1" , merge ( bistaTransfT1[["personpars"]], red, by = toM, all = TRUE))
-           stopifnot ( nrow(dat1) == nrow(bistaTransfT1[["personpars"]]))
-           dat2<- data.frame ( trend = "T2" , merge ( bistaTransfT2[["personpars"]], red, by = toM, all = TRUE))
-           stopifnot ( nrow(dat2) == nrow(bistaTransfT2[["personpars"]]))
-           if ( makeIdsUnique == TRUE ) {
-                dat1[, paste(idT1, "unique", sep="_")] <- paste(dat1[, "trend"], dat1[, idT1], sep="_")
-                dat2[, paste(idT1, "unique", sep="_")] <- paste(dat2[, "trend"], dat2[, idT1], sep="_")
-           }
-           return(rbind ( dat1, dat2))}
+prepRep <- function(calibT2, bistaTransfT1, bistaTransfT2, makeIdsUnique = TRUE){

```

Comment:
I didn't add anything new, because I think the arguments are already checked. Are there any additional checks needed?

Reply:
No, I don't think so. In the meantime, I think that `prepRep()` could also be removed from the package. It has not exactly proven to be useful and, as far as I know, is hardly used (or not at all, actually). It became obsolete with the introduction of eatGADS at the latest.

Reply:
I deleted the function and also its man page (.rd file).